app/src/main/java/com/github/libretube/workers/NotificationWorker.kt

```diff
@@ -125,47 +128,46 @@ class NotificationWorker(appContext: Context, parameters: WorkerParameters) :
+        val newStreams = applicationContext.resources
+            .getQuantityString(R.plurals.channel_new_streams, streams.size, streams.size)
+        val summary = NotificationCompat.InboxStyle()
+        streams.forEach {
+            summary.addLine(it.title)
+        }
+        val summaryNotification = createNotificationBuilder(group)
+            .setContentTitle(streams[0].uploaderName)
+            .setContentText(newStreams)
+            // The intent that will fire when the user taps the notification
+            .setContentIntent(pendingIntent)
+            .setGroupSummary(true)
+            .setStyle(summary)
+            .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_SUMMARY)
+            .build()
 
-            val notificationId = group.hashCode()
-            val intent = Intent(applicationContext, MainActivity::class.java)
-                .setFlags(INTENT_FLAGS)
-                .putExtra(IntentData.channelId, group)
-            val pendingIntent = PendingIntentCompat
-                .getActivity(applicationContext, notificationId, intent, FLAG_UPDATE_CURRENT, false)
-
-            // Create summary notification containing new streams for Android versions below 7.0.
-            val newStreams = applicationContext.resources
-                .getQuantityString(R.plurals.channel_new_streams, streams.size, streams.size)
-            val summary = NotificationCompat.InboxStyle()
-            streams.forEach {
-                summary.addLine(it.title)
-            }
-            val summaryNotification = createNotificationBuilder(group)
-                .setContentTitle(streams[0].uploaderName)
-                .setContentText(newStreams)
-                // The intent that will fire when the user taps the notification
-                .setContentIntent(pendingIntent)
-                .setGroupSummary(true)
-                .setStyle(summary)
-                .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_SUMMARY)
-                .build()
-
-            notificationManager.notify(notificationId, summaryNotification)
+        // Create stream notifications. These are automatically grouped on Android 7.0 and later.
+        val notificationsAndIds = withContext(Dispatchers.IO) {
+            streams.map { async { createStreamNotification(group, it) } }
+                .awaitAll()
+                .sortedBy { (_, uploaded, _) -> uploaded }
+                .map { (notificationId, _, notification) -> notificationId to notification }
+        } + (summaryId to summaryNotification)
+        notificationsAndIds.forEach { (notificationId, notification) ->
+            notificationManager.notify(notificationId, notification)
         }
     }
 
-    private suspend fun showStreamNotification(
+    private suspend fun createStreamNotification(
         group: String,
         stream: StreamItem,
-        isSingleNotification: Boolean,
-    ) {
+    ): Triple<Int, Long?, Notification> {
```

Comment:
I think returning a `Triple` is not really easy to read as the types `Int` and `Long` are very generic and don't really indicate what they are.
It might be a good idea to add comments about what all the returned values are or create a simple, private data class to become returned here.